pkg/proxy/integrations/httpparser/httpparser.go

```diff
@@ -489,11 +490,37 @@ func decodeOutgoingHttp(requestBuffer []byte, clientConn, destConn net.Conn, h *
 		if err != nil {
 			logger.Error("error while matching http mocks", zap.Error(err))
 		}
-
 		if !isMatched {
 			passthroughHost := false
-			for _, host := range models.PassThroughHosts {
-				if req.Host == host {
+			passThrough := models.PassThroughHosts
+			portPassThrough := []models.Filters{}
+			for _, filters := range h.GetProxyHost().Filters {
+				if filters.Port == 0 {
+					passThrough = append(passThrough, filters.Host, filters.Path)
+				} else {
+					portPassThrough = append(portPassThrough, filters)
+				}
+			}
+			// Define the regular expression pattern
+
+			for _, host := range passThrough {
+				regex, err := regexp.Compile(host)
+				if err != nil {
+					continue
+				}
+				subMatches := regex.FindStringSubmatch(req.URL.String())
+				if len(subMatches) > 0 && host != "" || req.Host == host {
+					passthroughHost = true
+				}
+
+			}
+			for _, filter := range portPassThrough {
```

Comment:
I think we shouldn't  use regex in case of only port

Reply:
this is checking path for a defined port, thus checking the endpoint with reger